### PR TITLE
Fix get_name_by_id, improve debug statements, exp. auto_reconnect

### PR DESCRIPTION
**Pull request by jstockdale**
This fixes get_name_by_id for hex values less than 0x10000000 by formatting the hex string using %08x (which ensures 8 bytes) and prevents the leading zero from getting narfed by the type conversions. This fixes "Unknown User" messages in the message_history window.

Add a lookup_button conditionally, which lets us get_name_by_id on any node that is currently selected (same as direct messages).

Improve debug statements to include hex_user_id when it's in scope.

Implement experimental auto_reconnect with delay.

### CODE REVIEW

**Comment by pdxlocations** [comment]
Nice fixes!  What's the purpose of the lookup button?

**Comment by jstockdale** [comment]
Oh it looks up the selected id in the sql lite database and if you have debug enabled you'll get output in your terminal showing if it found the id or not. It renders below the traceroute button and works the same way as the traceroute or direct_message buttons ... you select the node you care about or enter the id value manually, and in this case it calls get_id_by_name("long", ...) with the id you have selected.

Basically it makes it a lot easier to debug the id lookup (without directly connecting to the sql lite database), and I figured it might be useful if anyone else needs to do the same.

**Comment by pdxlocations** [comment]
> Oh it looks up the selected id in the sql lite database and if you have debug enabled you'll get output in your terminal showing if it found the id or not. It renders below the traceroute button and works the same way as the traceroute or direct_message buttons ... you select the node you care about or enter the id value manually, and in this case it calls get_id_by_name("long", ...) with the id you have selected.

I get this, but when would the selected id NOT be in your database?  Since you just clicked on the node id, why would you need to display the node id?

**Comment by jstockdale** [comment]
Hold on merging I have one fix I'm adding...

**Comment by jstockdale** [comment]
It should be in your database unless there's an error like a leading 0 that gets dropped 😂 

But also I just noticed there's a int conversion bug I need to fix let me update the pr. Want me to remove the Lookup button while I'm at it?
